src/query/service/src/sessions/query_ctx.rs

```diff
@@ -1190,6 +1190,13 @@ impl TableContext for QueryContext {
                 self.get_settings().get_enterprise_license(),
                 Feature::SystemHistory,
             )?;
+
+            if GlobalConfig::instance().log.history.is_invisible(table) {
+                return Err(ErrorCode::InvalidArgument(format!(
```

Comment:
UnknownTable is better?

Reply:
`UnknownTable` will be special handled. We just want to stop early here.